File: Exercism Codes/C#/CalculatorConundrum.cs
using System;

public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string operation)
    {
        try
        {
            return $"{operand1} {operation} {operand2} = " + (operation switch
            {
                "+" => SimpleOperation.Addition(operand1, operand2),
                "*" => SimpleOperation.Multiplication(operand1, operand2),
                "/" => SimpleOperation.Division(operand1, operand2),
                "" => throw new ArgumentException(),
                null => throw new ArgumentNullException(),
                _ => throw new ArgumentOutOfRangeException(),
            });
        }
        catch (DivideByZeroException)
        {
            return "Division by zero is not allowed.";
        }
    }
}

File: Exercism Codes/Java/Calculator Conundrum/CalculatorConundrum.java
class CalculatorConundrum {
    public String calculate(int operand1, int operand2, String operation) {
        if (operation == null) {
            throw new IllegalArgumentException("Operation cannot be null");
        }
        if (operation.isEmpty()) {
            throw new IllegalArgumentException("Operation cannot be empty");
        }
        String expression = operand1 + " " + operation + " " + operand2 + " = ";
        switch (operation) {
            case "+" -> {
                return expression + (operand1 + operand2);
            }
            case "*" -> {
                return expression + (operand1 * operand2);
            }
            case "/" -> {
                if (operand2 == 0) {
                    throw new IllegalOperationException("Division by zero is not allowed", new ArithmeticException());
                }
                return expression + (operand1 / operand2);
            }
            default -> throw new IllegalOperationException("Operation '" + operation + "' does not exist");
        }
    }
}